src/v2/Apps/Authentication/Utils/useAuthFormData.ts

```diff
@@ -0,0 +1,77 @@
+import { ModalOptions, ModalType } from "v2/Components/Authentication/Types"
+import { useRouter } from "v2/System/Router/useRouter"
+import { getENV } from "v2/Utils/getENV"
+import Cookies from "cookies-js"
+import { AuthenticationMetaProps } from "../Components/AuthenticationMeta"
+import { computeTitle } from "./computeTitle"
+
+interface UseAuthFormDataProps {
+  canonical: string
+  pageTitle: string
+  type: ModalType
+}
+
+export function useAuthFormData({
+  canonical,
+  pageTitle,
+  type,
+}: UseAuthFormDataProps) {
+  const { match } = useRouter()
+
+  const {
+    action,
+    afterSignUpAction,
+    contextModule,
+    copy: copyQueryParam,
+    destination,
+    intent,
+    kind,
+    objectId,
+  } = match.location.query as ModalOptions
+
+  const title = computeTitle({
+    copy: copyQueryParam,
+    intent,
+    pageTitle,
+  })
+
+  const redirectTo = getENV("AUTHENTICATION_REDIRECT_TO")
+  const signupReferer = getENV("AUTHENTICATION_REFERER")
```

Comment:
Wouldn't we use sharify for this?

Reply:
`getENV` uses sharify under the hood, and is the SSR friendly way to access.

Reply:
Wait — Sharify isn't SSR friendly? I thought that was the whole point of it?

Reply:
As far as I understand, since our JS is now fully isomorphic when you try to import `sd` it will look for the window and fail. So getENV will first check environment -- either localStorage or process.env on server, and then sharify on client. Old force wasn't executing JS isomorphically like we are now, it was mostly just rendering jade templates on the server with data passed in (from express). Now that we're using sharify much more widely (and without consideration of where things are executing) we needed a helper.